src/tbp/monty/frameworks/models/motor_policies.py

```diff
@@ -768,6 +790,157 @@ def sensor_rotation_relative_to_world(self, state: MotorSystemState) -> Any:
+
+        # Helpful to conceptualize these movements by considering a unit circle,
+        # scaled by the radius distance_from_center
+        # This image may be useful for intuition:
+        # https://en.wikipedia.org/wiki/Exsecant#/media/File:Circle-trig6.svg
+
+        # Rotate about a circle centered in fron of the agent's current
+        # position; TODO decide how to deal with "coliding with"/entering an
+        # object; ?could just rotate about a point on which the sensor is present
+        # Currently as a heuristic we rotate about a point 4x the desired distance;
+        # as this is typically 2.5cm, this would mean a circle with radius 10cm
+        distance_from_center = self._desired_object_distance * 4
+
+        rotation_degrees = 30  # 30 degrees at a time; note this amount is also used
+        # to eventually re-orient ourselves back to the original central point;
+        # TODO may want to consider trying smaller step-sizes; will
+        # be less efficient, but may be important for smaller/distant objects that
+        # otherwise get missed
+
+        if self._touch_search_rotation_degrees >= 720:
+            # Perform a random upward or downward movement along the surface of a
+            # sphere, with its centre fixed 10 cm in front of the agent
+            logging.debug("Trying random search for object")
+            if self.rng.uniform() < 0.5:
+                orientation = "vertical"
+                logging.debug("Orienting vertically")
+            else:
+                orientation = "horizontal"
+                logging.debug("Orienting horizontally")
+
+            rotation_degrees = self.rng.uniform(-180, 180)
+            logging.debug(f"Random orientation amount is : {rotation_degrees}")
+
+        elif (
+            self._touch_search_rotation_degrees >= 360
+            and self._touch_search_rotation_degrees < 720
+        ):
+            logging.debug("Trying vertical search for object")
+            orientation = "vertical"
+
+        else:
+            logging.debug("Trying horizontal search for object")
+            orientation = "horizontal"
+
+        # Move tangentally to the point we're facing, resulting in the agent's
+        # orienation about the central point changing as specified by rotation_deg,
+        # but now where the agent is no longer on the edge of the circle
+        move_lat_amount = np.tan(np.radians(rotation_degrees)) * distance_from_center
+        # The lateral movement will be down relative to the agent's facing position
+        # in the case of orient vertical, and left in the case of orient horizontal
+
+        # The below calculates the exsecant, which provides the necessary
+        # movement to bring the agent back to the same radius around the central
+        # point
+        move_forward_amount = distance_from_center * (
+            (1 - np.cos(np.radians(rotation_degrees)))
+            / np.cos(np.radians(rotation_degrees))
+        )
```

Comment:
If I'm understanding the math right, we're moving a little bit along the tangent line at the point of intersection with the circle, and then moving inward to correct for the mismatch between the circle arc and the tangent line.

Is this more efficient than just rotating around the center of the circle? Seems like a complicated approach to do the same thing.

_I realize this code is just copied from below, but it's a good a time as any to ask these questions_